tests/30rooms/01state.pl

```diff
@@ -242,14 +242,24 @@ sub get_test_state
 {
    my ( $user, $room_id, $type ) = @_;
 
-   matrix_initialsync_room( $user, $room_id, limit => 0 )->then( sub {
-      my ( $body ) = @_;
+   matrix_sync( $user,
+      filter            => '{"room":{"rooms":["'.$room_id.'"]},"account_data":{},"presence":{"types":[]}}',
+      update_next_batch => 0,
+      set_presence      => "offline",
+   )->then( sub {
+        my ( $body ) = @_;
 
-      my %state_by_type = partition_by { $_->{type} } @{ $body->{state} };
+        # We expect the event to be in the initial sync
+        sync_timeline_contains( $body, $room_id, sub {
+            my ( $event ) = @_;
+            return unless $event->{type} eq $type and $event->{state_key} eq $user->user_id;
 
-      my $event_id = $state_by_type{ $type }[0]{event_id};
+            log_if_fail "Event for ${\$user->user_id}", $event;
+            Future->done( $event->{event_id} );
```

Comment:
this creates a Future which is discarded?

Reply:
ah I think this should be at the higher level?

Reply:
Honestly I have no idea. I still don't think I grok how this works looking at it again.